src/lib/utilities/decode-payload.ts

```diff
@@ -69,42 +111,54 @@ export const convertPayloadToJson = async (
     }
   }
 
-  // Decode Search Attributes
-  if (anyAttributes?.searchAttributes?.indexedFields) {
-    const searchAttributes = anyAttributes?.searchAttributes?.indexedFields;
+  decodePayloadAttributes(anyAttributes);
 
-    Object.entries(searchAttributes).forEach(([key, value]) => {
-      searchAttributes[key] = decodePayload(value);
-    });
-  }
+  return anyAttributes;
+};
 
-  // Decode Memo
-  if (anyAttributes?.memo?.fields) {
-    const memo = anyAttributes?.memo?.fields;
+export const convertPayloadToJsonWithWebsocket = async (
+  eventAttribute: EventAttribute,
+  websocket?: DataConverterWebsocketInterface,
+): Promise<EventAttribute> => {
+  // This anyAttribues allows us to use ?. notation to introspect the object which is a safe access pattern.
+  // Because of the way we wrote our discrimited union we have to use this any. If we have two objects that
+  // don't have the same property TS won't let us access that object without verifying the type string like
+  // attributes.type === "ATypeWithInput/Result"
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const anyAttributes = eventAttribute as any;
+  const potentialPayload = (anyAttributes?.input?.payloads ??
+    anyAttributes?.result?.payloads ??
+    null) as Payload[] | null;
 
-    Object.entries(memo).forEach(([key, value]) => {
-      memo[key] = decodePayload(value);
-    });
-  }
+  if (potentialPayload) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let JSONPayload: string | Payload | Record<any, any>;
 
-  // Decode Header
-  if (anyAttributes?.header?.fields) {
-    const header = anyAttributes?.header?.fields;
+    const ws = websocket ?? dataConverterWebsocket;
 
-    Object.entries(header).forEach(([key, value]) => {
-      header[key] = decodePayload(value);
-    });
-  }
+    if (ws?.hasWebsocket) {
+      // Convert Payload data
+      const awaitData = await Promise.all(
+        (potentialPayload ?? []).map(
+          async (payload) =>
+            await convertPayloadWithWebsocket(payload, ws.websocket),
+        ),
+      );
 
-  // Decode Query Result
-  // This one is a best guess from the previous codebase and needs verified
-  if (anyAttributes?.queryResult) {
-    const queryResult = anyAttributes?.queryResult;
+      JSONPayload = awaitData;
+    } else {
+      JSONPayload = potentialPayload.map(decodePayload);
+    }
 
-    Object.entries(queryResult).forEach(([key, value]) => {
-      queryResult[key] = decodePayload(value);
-    });
+    if (anyAttributes?.input?.payloads) {
+      anyAttributes.input.payloads = JSONPayload;
+    }
+    if (anyAttributes?.result?.payloads) {
+      anyAttributes.result.payloads = JSONPayload;
+    }
   }
 
+  decodePayloadAttributes(anyAttributes);
```

Comment:
It may be nice to move this outside of either of these so the payload would be base64 decoded independently of the logic to convert with a websocket or remote codec.